app/code/Magento/PageCache/view/frontend/templates/inline_form_key_provider.phtml

```diff
@@ -0,0 +1,21 @@
+  
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/** @var \Magento\Framework\View\Element\Template $block */
+?>
+<script>
+var formKey = (function () {
+    
+    var value = "; " + document.cookie;
+    var parts = value.split("; " + 'form_key' + "=");
+    
+    if (parts.length == 2) {
+        return parts.pop().split(";").shift();
+    }
+    
+})();
+</script>
```

Comment:
1. As Magento introduced CSP - we need to use something like this:
https://github.com/magento/magento2/blob/89d3cda99975c0db2967165311184be84d26e9b4/app/code/Magento/Theme/view/frontend/templates/page/js/require_js.phtml#L15

2. Seems like case with missing form_key cookie isn't covered here
https://github.com/magento/magento2/blob/9544fb243d5848a497d4ea7b88e08609376ac39e/app/code/Magento/PageCache/view/frontend/web/js/page-cache.js#L120-L123

3. I think we need to remove following code as we basically replacing it with inline implementation https://github.com/magento/magento2/blob/9544fb243d5848a497d4ea7b88e08609376ac39e/app/code/Magento/PageCache/view/frontend/web/js/page-cache.js#L300-L303